gnrc_ipv6_ext_frag: add configuration option to keep oldest entry

miri64 opened the pull request:
<!--
The RIOT community cares a lot about code quality.
Therefore, before describing what your contribution is about, we would like
you to make sure that your modifications are compliant with the RIOT
coding conventions, see https://github.com/RIOT-OS/RIOT/wiki/Coding-conventions.
-->

### Contribution description
This gives a similar treatment to the reassembly buffer of `gnrc_ipv6_ext_frag` as I already introduced for `gnrc_sixlowpan_frag_rb`: a configuration option to turn of the overriding of the oldest entry when the reassembly buffer is full and a new incoming datagram tries to be reassembled.
<!--
Put here the description of your contribution:
- describe which part(s) of RIOT is (are) involved
- if it's a bug fix, describe the bug that it solves and how it is solved
- you can also give more information to reviewers about how to test your changes
-->


### Testing procedure
`tests/gnrc_ipv6_ext_frag` should still compile and the test should still be running.

If you want to test the new option, try compiling `tests/gnrc_udp` `-DCONFIG_GNRC_IPV6_EXT_FRAG_RBUF_DO_NOT_OVERRIDE` and do e.g. [Task 4.10 of the release specs](https://github.com/RIOT-OS/Release-Specs/blob/master/04-single-hop-6lowpan-icmp/04-single-hop-6lowpan-icmp.md#task-10-exprimental---icmpv6-echo-with-large-payload-ipv6-fragmentation).
<!--
Details steps to test your contribution:
- which test/example to compile for which board and is there a 'test' command
- how to know that it was not working/available in master
- the expected success test output
-->


### Issues/PRs references
None
<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

leandrolanzieri commented on sys/net/gnrc/network_layer/ipv6/ext/frag/Kconfig:
```diff
@@ -40,4 +40,13 @@ config GNRC_IPV6_EXT_FRAG_RBUF_TIMEOUT_US
+    help
+        When not set, it will cause the reassembly buffer to
+        override the oldest entry no matter what. When set, only the oldest
+        entry that is older than @ref CONFIG_GNRC_IPV6_EXT_FRAG_RBUF_DO_NOT_OVERRIDE
```
```suggestion
        entry that is older than @ref CONFIG_GNRC_IPV6_EXT_FRAG_RBUF_TIMEOUT_US
```

miri64 replied:
Oops!.

miri64 commented:
Added another fixup in line with https://github.com/RIOT-OS/RIOT/pull/13245#discussion_r373458975. Also changed the second sentence, as it advertised different behavior than is actually implemented (the IPv6 reassembly buffer works there a bit different from the 6LoWPAN reassembly buffer.

miri64 commented:
Squashed
